### PR TITLE
V14: Add additional validation around manipulating groups

### PR DESCRIPTION
**Pull request by nikolajlauridsen**
Adds additional validation around manipulating user groups, more specifically it ensures that the admin group is never left empty, since this would mean that no user could ever be added to the admin group again. 

Most of this is pretty run-of-the-mill, however, there was a little snafu with the `UpdateUserGroupsUserController`, since this more or less acts more like a UserGroup controller, I've left a clarifying comment in the top of the file.

## Testing

Ensure that you are not allowed to remove all users from the admin group.

### CODE REVIEW

File: src/Umbraco.Core/Services/UserGroupService.cs [hunk 1]
```diff
@@ -178,16 +178,33 @@ public async Task<Attempt<UserGroupOperationStatus>> DeleteAsync(ISet<Guid> keys
...
+        if (userGroupKeys.Contains(Constants.Security.AdminGroupKey) is false)
+        {
+            IUser[] usersToDeAdmin = users.Where(x => x.IsAdmin()).ToArray();
+            if (usersToDeAdmin.Length > 0)
+            {
+                // Unfortunately we have to resolve the admin group to ensure that it would not be left empty.
+                IUserGroup? adminGroup = await GetAsync(Constants.Security.AdminGroupKey);
+                if (adminGroup is not null && adminGroup.UserCount <= usersToDeAdmin.Length)
+                {
+                    return UserGroupOperationStatus.AdminGroupCannotBeEmpty;
```
Comment (bergmania) [inline src/Umbraco.Core/Services/UserGroupService.cs:203]: Remember to complete the scope before return. This is a valid use case and an exception and we do not have to rollback anything.

File: src/Umbraco.Core/Services/UserService.cs [hunk 2]
```diff
@@ -912,13 +912,24 @@ private async Task<UserOperationStatus> ValidateUserCreateModel(UserCreateModel
             return Attempt.FailWithStatus<IUser?, UserOperationStatus>(UserOperationStatus.MissingUser, existingUser);
         }
 
-        var userGroups = _userGroupRepository.GetMany().Where(x=>model.UserGroupKeys.Contains(x.Key)).ToHashSet();
+        IEnumerable<IUserGroup> allUserGroups = _userGroupRepository.GetMany().ToArray();
```
Comment (bergmania) [inline src/Umbraco.Core/Services/UserService.cs:915]: I know most of the returns in this method is not new in this PR, but same comment as before. We need to complete the scope, if nothing should be rolled back.

**Review by bergmania — CHANGES_REQUESTED** [review]

File: src/Umbraco.Core/Services/UserService.cs (hunk 2 above)
Comment (nikolajlauridsen) [reply src/Umbraco.Core/Services/UserService.cs:916]: Very good point 😄 I've completed the scopes where it's safe to do so in the 2 methods 😄

**Review by bergmania — APPROVED** [review]
Tests out good

**Pull request merged**